### ISSUE TITLE
upserting with long setter hash fails (postgres)

### ISSUE THREAD

**Issue by eckley** [seamusabshere/upsert#21]
Trying to upsert a database table with many columns fails while the same code with some items removed succeeds. 
Simplified code:

``` ruby
Upsert.batch(connection,table_name) do |upsert|
          nereus_update_hash.each do |item|

            update_row = {:credit=>475358, :daily_credit=>20, :monthly_network_usage=>841123927, :online_now=>2, :online_today=>2, :mips_now=>3051, :mips_today=>3051, :active=>1, :updated_at=>Time.now, :created_at=>Time.now}
            upsert.row({:nereus_id => item[0].to_i}, update_row)
          end
        end
```

This fails with the error, the table name is nereus_stats_items:

```
ERROR:  function upsert_nereus_stats_items_sel_nereus_id_set_active_a_2167893291(unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown) is not unique
LINE 1: SELECT upsert_nereus_stats_items_SEL_nereus_id_SET_active_A_...
               ^
HINT:  Could not choose a best candidate function. You might need to add explicit type casts.
```

however change update_row to either and the script runs fine:

``` ruby
update_row = {:monthly_network_usage=>841123927, :online_now=>2, :online_today=>2, :mips_now=>3051, :mips_today=>3051, :active=>1, :updated_at=>Time.now, :created_at=>Time.now}
update_row = {:credit=>475358, :daily_credit=>20, :online_now=>2, :online_today=>2, :mips_now=>3051, :mips_today=>3051, :active=>1, :updated_at=>Time.now, :created_at=>Time.now}
```

I'm using:
ruby 1.9.3p194
rails 3.2.11
PostgreSQL 9.1.9 on x86_64-unknown-linux-gnu, compiled by gcc (Ubuntu/Linaro 4.7.2-2ubuntu1) 4.7.2, 64-bit
upsert 1.2.0

Any help as to why its doing this would be greatly appreciated.
Cheers
Alex

**Comment by fatcatt316** [comment]
I just ran into this myself... I'm importing about half a million records, and this pops up very rarely.  I'm on Ruby 2.2 with PostgreSQL (pg-0.18.1 is the gem) and using upsert 2.0.4.  Any clues about this?

    ERROR:  function upsert2_0_4_fruits_sel_store_id_set_flavor_a_890319843(unknown, unknown, unknown, unknown, unknown, unknown, u
nknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, u
nknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, unknown, u
nknown, unknown, unknown, unknown, unknown, unknown, unknown) is not unique
LINE 1: SELECT upsert2_0_4_fruits_SEL_store_id_SET_flavor_A_...
               ^
HINT:  Could not choose a best candidate function. You might need to add explicit type casts.

**Comment by jufemaiz** [comment]
This is still an issue :\

**Comment by jufemaiz** [comment]
@eckley +@seamusabshere have you got any thoughts?

**Comment by seamusabshere** [comment]
@jufemaiz does it happen consistently after you delete all upsert functions?

**Comment by jufemaiz** [comment]
@seamusabshere to date it has. I've done another cleanout and will see how the next few weeks go.

**Comment by jufemaiz** [comment]
@seamusabshere starting to see the issue arise again :(

```
ERROR:  function upsert2_1_0_measurement_data_sel_measurement_point_id2088041071(unknown, unknown, unknown, unknown, unknown, unknown) is not unique
LINE 1: SELECT upsert2_1_0_measurement_data_SEL_measurement_point_id...
               ^
HINT:  Could not choose a best candidate function. You might need to add explicit type casts.
```

**Comment by jufemaiz** [comment]
```ruby
ERROR:  function upsert2_1_0_measurement_data_sel_measurement_point_id2088041071(unknown, unknown, unknown, unknown, unknown, unknown) is not unique
LINE 1: SELECT upsert2_1_0_measurement_data_SEL_measurement_point_id...
               ^
HINT:  Could not choose a best candidate function. You might need to add explicit type casts.
```

Still happening, and in this instance there's only one (1) function matching

```regexp
/^upsert/
```

**Comment by pnomolos** [comment]
Marking this as closed - I believe it's fixed via #89.
